cmake/modules/Findcrypto.cmake

```diff
@@ -56,7 +56,7 @@ else()
     )
 
     if (NOT crypto_LIBRARY)
-        if (BUILD_SHARED_LIBS)
+        if (BUILD_SHARED_LIBS OR ${S2N_BUILD_SHARED_LIBS})
```

Comment:
Discussed offline that we should use `${*}` syntax: https://cmake.org/cmake/help/latest/command/if.html#variable-expansion